### ISSUE TITLE
Evaluation tests failing after qp update

### ISSUE THREAD

**Issue by sschmidt23** [LSSTDESC/rail_attic#330]
On creating a recent PR today, I saw two of the test_evaluation.py tests are now failing.  I see a new release for qp was created yesterday with some changes that we apparently did not keep up with in a couple of the evaluator stages, we should track those down and fix them.  Problems are namely: 
1) test_pit_metrics calls `rail.evaluation.metrics.pit.PITAD`, which has 
`pit = qp.Ensemble(qp.quant_piecewise, data=dict(quants=eval_grid, locs=np.atleast_2d(data_quants)))`
causing error
`E       AttributeError: module 'qp' has no attribute 'quant_piecewise'`
2) test_evaluation_stage's `evaluator.evaluate(pdf, truth)` tries to evaluate a CDF and sees multiple x values with the same (0.0) value, and make_interp_spline throws exception:
`E           ValueError: Expect x to not have duplicates`

**Comment by sschmidt23** [comment]
It looks like evaluator.evaluate calls a bunch of the deprecated functions that were moved to qp.  Should we clean this up a bit, just remove those deprecated functions and update the rail.evaluation.evaluator.py stuff to call the qp functions?  Or, was there some reason we left things as deprecated and didn't do this earlier?  @drewoldag do you remember why you decided to do things this way?

**Comment by sschmidt23** [comment]
Actually, do we even need the Evaluator class in rail.evaluation.evaluator.py anymore?  Could we just drop it altogether, and just call the qp stuff directly in the goldenspike notebook?  I think that might be the only place it's used (well that and the demo notebook in examples/evaluation)

**Comment by eacharles** [comment]
The class let’s you put it in a pipeline…-eOn Apr 11, 2023, at 5:24 PM, Sam Schmidt ***@***.***> wrote:﻿
Actually, do we even need the Evaluator class in rail.evaluation.evaluator.py anymore?  Could we just drop it altogether, and just call the qp stuff directly in the goldenspike notebook?  I think that might be the only place it's used (well that and the demo notebook in examples/evaluation)

**Comment by eacharles** [comment]
My vote would be to call the qp functions….-eOn Apr 11, 2023, at 5:24 PM, Sam Schmidt ***@***.***> wrote:﻿
Actually, do we even need the Evaluator class in rail.evaluation.evaluator.py anymore?  Could we just drop it altogether, and just call the qp stuff directly in the goldenspike notebook?  I think that might be the only place it's used (well that and the demo notebook in examples/evaluation)

**Comment by sschmidt23** [comment]
Ah, true, having the functions easily pipeline-able is nice. I'll look at swapping out the functions tomorrow.

**Comment by aimalz** [comment]
Re: Evaluator, it's also necessary for the metrics that don't belong in qp, i.e. any that require information beyond the PDFs themselves. That would include the CDE loss and any others that account for the photometry (probably the guts of the Cal-PIT, SOM QC, and other "recalibration" approaches really belong in rail.evaluation, too) as well as those involving point estimates (even if qp can be used to extract those from the PDFs once specified). But rail.evaluation is indeed underdeveloped relative to the other subpackages so probably merits a thorough audit before v1.

**Comment by drewoldag** [comment]
@sschmidt23 To you question about why I chose to leave the deprecated methods in pit.py. 

Basically I tried to find references to the PIT and Brier methods in RAIL and updated those, but I figured there might be some that I missed. 

That's in this PR (which you've probably already seen): https://github.com/LSSTDESC/RAIL/pull/283

I used the RAIL implementation of PIT and Brier in the tests (even though they are marked as deprecated) because it doesn't make sense to have unit tests in RAIL to test code in qp. I implemented/ported unit tests to qp for these as well. 

I am in favor of removing the deprecated methods and class eventually, my recommendation would be to remove the implementation and replace it with a raised exception or a logger warning directing the user to switch to the qp implementation. And then at a later date (after v1.0?) remove the methods entirely.

But if it makes sense to just yank them out now, then that's cool too :)
